ravenframework/CodeInterfaceClasses/Generic/GenericCodeInterface.py

```diff
@@ -178,3 +188,28 @@ def createNewInput(self,currentInputFiles,origInputFiles,samplerType,**Kwargs):
     parser.modifyInternalDictionary(**Kwargs)
     parser.writeNewInput(infiles,origfiles)
     return currentInputFiles
+
+  def onlineStopCriteria(self, command, output, workDir):
+    """
+      This method is called by RAVEN during the simulation.
+      It is intended to provide means for the code interface to monitor the execution of a run
+      and stop it if certain criteria are met (defined at the code interface level)
+      For example, the underlying code interface can check for a figure of merit in the output file
+      produced by the driven code and stop the simulation if that figure of merit is outside a certain interval
+      (e.g. Pressure > 2 bar, stop otherwise, continue).
+      If the simulation is stopped because of this check, the return code is set artificially to 0 (normal termination) and
+      the 'checkForOutputFailure' method is not called. So the simulation is considered to be successful.
+
+      @ In, command, string, the command used to run the just ended job
+      @ In, output, string, the Output name root
+      @ In, workingDir, string, current working dir
+      @ Out, stopSim, bool, True if the job needs to stop being executed, False if it needs to continue to be executed
+    """
+    stopSim = False
+    if self.stoppingCriteriaFunction is not None:
+      outCsv = os.path.join(workDir,output+".csv")
+      if os.path.exists(outCsv):
+        df = pd.read_csv(outCsv)
+        results =  dict((header, np.array(df[header])) for header in df.columns)
+        stopSim = self.stoppingCriteriaFunction.evaluate(self.stoppingCriteriaFunction.name,results)
```

Comment:
@alfoa FYI, we observed a failure on windows for the testing of this capability. The reason is: the output is created, but no data has been written yet, which means df[header] is empty, but the stoppingCriteriaFunction got evaluated with empty data which causes the failure of some jobs. We can either fix it here or update the stoppingCriteriaFunction to check it. Do you have some suggestions?